challenge-react/src/Card.js

```diff
@@ -0,0 +1,234 @@
+    opacity: 0.95;
+    animation: fadeIn 0.5s;
+    .card__showDonate {
+      top: 40%;
+      left: 50%;
+      transform: translateX(-50%);
+      position: relative;
+      width: fit-content;
+      display: flex;
+      flex-direction: column;
+    }
+    .card__closedDonate {
+      display: block;
+      width: 16px;
+      height: 16px;
+      opacity: 0.3;
+      top: 10px;
+      right: 15px;
+      cursor: pointer;
+      position: absolute;
+      &:hover {
+        opacity: 1;
+      }
+      &:before,
+      &:after {
+        position: absolute;
+        left: 7px;
+        content: " ";
+        height: 16px;
+        width: 2px;
+        background-color: #333;
+      }
+      &:before {
+        transform: rotate(45deg);
+      }
+      &:after {
+        transform: rotate(-45deg);
+      }
+    }
+  }
+
+  .card__buttonPay {
+    text-align: center;
+    height: 25px;
+    font-weight: 500;
+    background: transparent;
+    color: #5284ee;
+    border: 2px solid #517af1;
+    border-radius: 6px;
+    width: 65px;
+    cursor: pointer;
+    margin-left: auto;
+    margin-right: auto;
+  }
+  .card__textSelectDonate,
+  .card__boxSelectPay {
+    margin-bottom: 10px;
+    text-align: center;
+  }
+  .card__allDonate {
+    position: absolute;
+    bottom: 10px;
+    left: 50%;
+    transform: translateX(-50%);
+    width: fit-content;
+  }
+
+  .card__containerRadio {
+    display: inline-block;
+    position: relative;
+    margin-left: 0px;
+    padding-left: 32px;
+    width: fit-content;
+    margin-right: 10px;
+    margin-bottom: 12px;
+    cursor: pointer;
+    font-size: 16px;
+    user-select: none;
+
+    &:hover input ~ .checkmark {
+      background-color: #ccc;
+    }
+    input {
+      position: absolute;
+      opacity: 0;
+      cursor: pointer;
+      &:checked ~ .checkmark {
+        background-color: #2196f3;
+        &:after {
+          display: block;
+        }
+      }
+    }
+    .checkmark {
+      position: absolute;
+      top: 0;
+      left: 0;
+      height: 25px;
+      width: 25px;
+      background-color: #eee;
+      border-radius: 50%;
+      &:after {
+        content: "";
+        position: absolute;
+        display: none;
+        top: 9px;
+        left: 9px;
+        width: 8px;
+        height: 8px;
+        border-radius: 50%;
+        background: white;
+      }
+    }
+  }
+`;
+
+const Card = (props) => {
+  const {
+    item = {},
+    selectedAmount,
+    setSelectedAmount,
+    showDonate = 0,
+    setShowDonate,
+    dataPayment = [],
+    handlePay,
+  } = props;
+
+  const allDonate = dataPayment.reduce((sum, number) => {
+    return sum + number.amount;
+  }, 0);
+
+  const openShowDonate = (val) => {
+    setShowDonate(val);
+    setSelectedAmount(10);
+  };
+
+  const closeShowDonate = () => {
+    setShowDonate(0);
+    setSelectedAmount(10);
+  };
+
+  const payments = [10, 20, 50, 100, 500].map((amount, j) => (
+    <label key={j} className="card__containerRadio">
```

Comment:
In this scenario, using the index as the key here is not a bad idea. However, React doesn't recommend using the index as the key. It may impact performance.

How about using the amount here as the key?